src/zip/checksum_reader.cr

```diff
@@ -0,0 +1,32 @@
+module Zip
+  # Computes a CRC32 while reading from an underlying IO,
+  # optionally verifying the computed value against an
+  # expected one.
+  private class ChecksumReader
```

Comment:
We could do that, maybe in a next PR. I checked some other implementations, for example Go has something similar and it's private to the zip package, maybe it's not that useful in other contexts.